src/main/groovy/lang/IntRange.java

```diff
@@ -379,32 +379,25 @@ public boolean containsAll(Collection other) {
     @Override
     public void step(int step, Closure closure) {
         if (step == 0) {
-            if (!getFrom().equals(getTo())) {
-                throw new GroovyRuntimeException("Infinite loop detected due to step size of 0");
-            }
-            return; // from == to and step == 0, nothing to do, so return
+            return;
```

Comment:
I don't mind either, but this is what @blackdrag requested.
Please decide and let me know.